binary_port/src/error_code.rs

```diff
@@ -87,7 +269,109 @@ impl fmt::Display for UnknownErrorCode {
 impl std::error::Error for UnknownErrorCode {}
 
 impl From<InvalidTransaction> for ErrorCode {
-    fn from(_value: InvalidTransaction) -> Self {
-        ErrorCode::InvalidTransaction
+    fn from(value: InvalidTransaction) -> Self {
+        match value {
+            InvalidTransaction::Deploy(invalid_deploy) => handle_invalid_deploy(invalid_deploy),
+            InvalidTransaction::V1(invalid_transaction) => {
+                handle_invalid_transaction(invalid_transaction)
+            }
+            _ => ErrorCode::InvalidTransactionOrDeployUnspecified,
+        }
+    }
+}
+
+fn handle_invalid_deploy(invalid_deploy: InvalidDeploy) -> ErrorCode {
+    match invalid_deploy {
+        InvalidDeploy::InvalidChainName { .. } => ErrorCode::InvalidDeployChainName,
+        InvalidDeploy::DependenciesNoLongerSupported => {
+            ErrorCode::InvalidDeployDependenciesNoLongerSupported
+        }
+        InvalidDeploy::ExcessiveSize(_) => ErrorCode::InvalidDeployExcessiveSize,
+        InvalidDeploy::ExcessiveTimeToLive { .. } => ErrorCode::InvalidDeployExcessiveTimeToLive,
+        InvalidDeploy::TimestampInFuture { .. } => ErrorCode::InvalidDeployTimestampInFuture,
+        InvalidDeploy::InvalidBodyHash => ErrorCode::InvalidDeployBodyHash,
+        InvalidDeploy::InvalidDeployHash => ErrorCode::InvalidDeployHash,
+        InvalidDeploy::EmptyApprovals => ErrorCode::InvalidDeployEmptyApprovals,
+        InvalidDeploy::InvalidApproval { .. } => ErrorCode::InvalidDeployApproval,
+        InvalidDeploy::ExcessiveSessionArgsLength { .. } => {
+            ErrorCode::InvalidDeployExcessiveSessionArgsLength
+        }
+        InvalidDeploy::ExcessivePaymentArgsLength { .. } => {
+            ErrorCode::InvalidDeployExcessivePaymentArgsLength
+        }
+        InvalidDeploy::MissingPaymentAmount => ErrorCode::InvalidDeployMisssingPaymentAmount,
+        InvalidDeploy::FailedToParsePaymentAmount => {
+            ErrorCode::InvalidDeployFailedToParsePaymentAmount
+        }
+        InvalidDeploy::ExceededBlockGasLimit { .. } => {
+            ErrorCode::InvalidDeployExceededBlockGasLimit
+        }
+        InvalidDeploy::MissingTransferAmount => ErrorCode::InvalidDeployMissingTransferAmount,
+        InvalidDeploy::FailedToParseTransferAmount => {
+            ErrorCode::InvalidDeployFailedToParseTransferAmount
+        }
+        InvalidDeploy::InsufficientTransferAmount { .. } => {
+            ErrorCode::InvalidDeployInsufficientTransferAmount
+        }
+        InvalidDeploy::ExcessiveApprovals { .. } => ErrorCode::InvalidDeployExcessiveApprovals,
+        InvalidDeploy::UnableToCalculateGasLimit => {
+            ErrorCode::InvalidDeployUnableToCalculateGasLimit
+        }
+        InvalidDeploy::UnableToCalculateGasCost => ErrorCode::InvalidDeployUnableToCalculateGasCost,
+        _ => ErrorCode::InvalidDeployUnspecified,
+    }
+}
+
+fn handle_invalid_transaction(invalid_transaction: InvalidTransactionV1) -> ErrorCode {
```

Comment:
similarly to the above, I'd make this `impl From<InvalidTransactionV1> for ErrorCode`